### PR TITLE
Implement input map_to_region command

### PR DESCRIPTION
**Pull request by tadeokondrak**
This has been in the documentation for a while but never got implemented.

### CODE REVIEW

File: sway/sway-input.5.scd [hunk 1]
```diff
@@ -88,7 +88,7 @@ The following commands may only be used in the configuration file.
 	Maps inputs from this device to the specified output. Only meaningful if the
 	device is a pointer, touch, or drawing tablet device.
 
-*input* <identifier> map_to_region <WxH@X,Y>
+*input* <identifier> map_to_region <X> <Y> <width> <height>
```
Comment (tadeokondrak) [inline sway/sway-input.5.scd]: It seems like you can't have commas in command invocations, so I had to change this. It's similar to `grim`'s output format, but with spaces as all separators.

File: sway/commands/input/map_to_region.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	if ((error = checkarg(argc, "map_to_region", EXPECTED_EQUAL_TO, 4))) {
+		return error;
+	}
+	struct input_config *ic = config->handler_context.input_config;
+	if (!ic) {
+		return cmd_results_new(CMD_FAILURE, "No input device defined");
+	}
+
+	// This is used to clear the current output mapping.
+	ic->mapped_to_output = strdup("");
```
Comment (ddevault) [inline sway/commands/input/map_to_region.c]: This is weird. Can we set it to NULL and then update other code to not segfault? Do we have to free the old string?

File: sway/input/seat.c [hunk 3]
```diff
@@ -576,19 +576,39 @@ static void seat_reset_input_config(struct sway_seat *seat,
...
+	struct wlr_box *mapped_to_region = NULL;
 
 	struct input_config *ic = input_device_get_config(
 			sway_device->input_device);
 	if (ic != NULL) {
 		sway_log(SWAY_DEBUG, "Applying input config to %s",
 			sway_device->input_device->identifier);
 
+		// We use an empty string as a marker to clear the mapped_to_output
+		// property, because a NULL set in a handler_context isn't preserved.
```
Comment (ddevault) [inline sway/input/seat.c]: Oh. This is still weird, though. Can it be fixed?

**Comment by ddevault** [comment]
This looks pretty good.

**Pull request merged**